dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AbstractGistReadOnlyController.java

```diff
@@ -94,47 +103,65 @@
     public @ResponseBody ResponseEntity<JsonNode> getObjectGist(
         @PathVariable( "uid" ) String uid,
-        HttpServletRequest request, HttpServletResponse response )
+        GistParams params )
         throws NotFoundException
     {
-        return gistToJsonObjectResponse( uid, createGistQuery( request, getEntityClass(), GistAutoType.L )
+        return gistToJsonObjectResponse( uid, createGistQuery( params, getEntityClass(), GistAutoType.L )
             .withFilter( new Filter( "id", Comparison.EQ, uid ) ) );
     }
 
-    @GetMapping( value = "/{uid}/gist", produces = "text/csv" )
+    @OpenApi.Response( String.class )
+    @GetMapping( value = { "/{uid}/gist", "/{uid}/gist.csv" }, produces = "text/csv" )
     public void getObjectGistAsCsv( @PathVariable( "uid" ) String uid,
-        HttpServletRequest request, HttpServletResponse response )
+        GistParams params, HttpServletResponse response )
         throws IOException
     {
-        gistToCsvResponse( response, createGistQuery( request, getEntityClass(), GistAutoType.L )
+        gistToCsvResponse( response, createGistQuery( params, getEntityClass(), GistAutoType.L )
             .withFilter( new Filter( "id", Comparison.EQ, uid ) )
             .toBuilder().typedAttributeValues( false ).build() );
     }
 
+    @Value
+    @OpenApi.Shared( value = false )
+    private static class GistListResponse
+    {
+        GistPager pager;
+
+        @SuppressWarnings( "java:S116" )
+        ObjectNode[] path$ = null;
+    }
+
+    @OpenApi.Response( { GistListResponse.class, ObjectNode[].class } )
     @GetMapping( value = "/gist", produces = APPLICATION_JSON_VALUE )
     public @ResponseBody ResponseEntity<JsonNode> getObjectListGist(
-        HttpServletRequest request, HttpServletResponse response )
+        GistParams params, HttpServletRequest request )
     {
-        return gistToJsonArrayResponse( request, createGistQuery( request, getEntityClass(), GistAutoType.S ),
+        return gistToJsonArrayResponse( request, createGistQuery( params, getEntityClass(), GistAutoType.S ),
             getSchema() );
     }
 
-    @GetMapping( value = "/gist", produces = "text/csv" )
-    public void getObjectListGistAsCsv( HttpServletRequest request, HttpServletResponse response )
+    @OpenApi.Response( value = String.class )
+    @GetMapping( value = { "/gist", "/gist.csv" }, produces = "text/csv" )
+    public void getObjectListGistAsCsv( GistParams params, HttpServletResponse response )
         throws IOException
     {
-        gistToCsvResponse( response, createGistQuery( request, getEntityClass(), GistAutoType.S )
+        gistToCsvResponse( response, createGistQuery( params, getEntityClass(), GistAutoType.S )
             .toBuilder().typedAttributeValues( false ).build() );
     }
 
+    @OpenApi.Response( status = BAD_REQUEST, value = WebMessage.class )
+    @OpenApi.Response( status = NOT_FOUND, value = WebMessage.class )
```

Comment:
Speculative comment, should we have a global way way of defining the behavior for 400, 404, 409? Seems to be the same for many endpoints?

Reply:
Maybe. I was mostly thinking about using this to provide better descriptions as to why this might happen. But so far I did not add such texts to them. But that was the plan.

Reply:
Right. Not a big issue. Can imagine the reason for 404 is probably mostly the same, and starting to explain 409 might be a lot of text. All good.